tests/lib/rules/require-reduce-negation.ts

```diff
@@ -0,0 +1,200 @@
+import { RuleTester } from "eslint"
+import rule from "../../../lib/rules/require-reduce-negation"
+
+const tester = new RuleTester({
+    parserOptions: {
+        ecmaVersion: "latest",
+        sourceType: "module",
+    },
+})
+
+tester.run("require-reduce-negation", rule as any, {
+    valid: [
+        String.raw`/[[abc]]/v`,
+        String.raw`/[\d]/u`,
+        String.raw`/[^\d]/v`, // Converting to `\D` does not reduce negation, so ignore it. The `negation` rule handles it.
+        String.raw`/[a--b]/v`,
+        String.raw`/[a&&b]/v`,
+        String.raw`/[^ab]/v`,
+        String.raw`/[^a&&b]/v;`,
+        String.raw`/[\s\p{ASCII}]/u`,
+        String.raw`/[^\S\P{ASCII}]/u`,
+        String.raw`/[^[]]/v`,
+        String.raw`/[a&&b&&[c]]/v`,
+        String.raw`/[a--b--[c]]/v`,
+    ],
+    invalid: [
+        {
+            code: String.raw`/[^[^abc]]/v`,
+            output: String.raw`/[abc]/v`,
+            errors: [
+                "This character class can be double negation elimination.",
+            ],
+        },
+        {
+            code: String.raw`/[^\D]/u`,
+            output: String.raw`/[\d]/u`,
+            errors: [
+                "This character class can be double negation elimination.",
+            ],
+        },
+        {
+            code: String.raw`/[^\P{ASCII}]/u`,
```

Comment:
Could you please add a test of this, but with the `i` flag enabled? I want to test it know that `/[\p{ASCII}]/iu` != `/[^\P{ASCII}]/iu`